src/Components/Endpoints/src/FormMapping/FormDataMapper.cs

```diff
@@ -2,31 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Components.Endpoints.FormMapping;
 
-internal static class FormDataMapper
+internal static partial class FormDataMapper
 {
     [RequiresDynamicCode(FormMappingHelpers.RequiresDynamicCodeMessage)]
     [RequiresUnreferencedCode(FormMappingHelpers.RequiresUnreferencedCodeMessage)]
     public static T? Map<T>(
         FormDataReader reader,
         FormDataMapperOptions options)
     {
+        FormDataConverter<T>? converter;
         try
         {
-            var converter = options.ResolveConverter<T>();
-            if (converter.TryRead(ref reader, typeof(T), options, out var result, out _))
+            converter = options.ResolveConverter<T>();
+            if (converter == null)
```

Comment:
Should we update the nullability of `FormDataMapperOptions.ResolverConverter` to indicate that it may return null?

Reply:
I've updated it